frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -78,17 +93,24 @@ export const useTourValuesForContext = (): TourContextType => {
     (state: RootState) => getRequiredFlagsByTour(state, selectorSteps),
     isEqual,
   );
-  const { completed } = getTourLocalStorageForPerspective(activePerspective);
+  const [perspective, setPerspective] = useState<string>(activePerspective);
+  const { completed } = getTourLocalStorageForPerspective(perspective);
   const onComplete = () => {
     if (completed === false) {
-      setTourCompletionLocalStorageDataForPerspective(activePerspective, true);
+      setTourCompletionLocalStorageDataForPerspective(perspective, true);
     }
   };
   const [tourState, tourDispatch] = useReducer<TourReducer>(tourReducer, {
     completedTour: completed,
     stepNumber: 0,
     startTour: !completed,
   });
+
+  useEffect(() => {
+    tourDispatch({ type: TourActions.initialize, payload: { completed } });
+    setPerspective(activePerspective);
+  }, [activePerspective, completed]);
```

Comment:
I don't think we need to run the effect on completed. We want to only run the effect when the perspective change so that the tour state can be initialized.

```suggestion
  }, [activePerspective]);
```

Reply:
updated dependency and disabled es lint rules for the line